test/integ_tests/gate_model_device_testing_utils.py

```diff
@@ -443,3 +441,15 @@ def run_circuit(circuit):
         assert np.allclose(result.measurement_probabilities["00"], 0.5, **tol)
         assert np.allclose(result.measurement_probabilities["11"], 0.5, **tol)
         assert len(result.measurements) == shots
+
+
+def batch_bell_pair_testing(device: AwsDevice, run_kwargs: Dict[str, Any]):
```

Comment:
Can we add an extra assert in here testing that we can still call task.result() on individual tasks after batch execute is done?

Reply:
Since the tasks are Mock objects, this wouldn't be particularly useful as the unit test; it belongs more in the integ test